src/bentoml/_internal/service/service.py

```diff
@@ -34,12 +35,23 @@
     from ..bento import Bento
     from ...grpc.v1 import service_pb2_grpc as services
     from .openapi.specification import OpenAPISpecification
+
+    APIDictBase = UserDict[str, InferenceAPI]
 else:
     grpc, _ = import_grpc()
+    APIDictBase = UserDict
 
 logger = logging.getLogger(__name__)
 
 
+class APIDict(APIDictBase):
+    def __getattr__(self, name):
+        try:
+            return self.__getitem__(name)
+        except KeyError:
+            raise AttributeError(f"InferenceAPI '{attr}' not found")
```

Comment:
Is this purely for convenience?

I think that asking people to write `svc.apis['encode']` is also fine

Even though `svc.apis.encode` is nice.

Reply:
yes just for convenience, although I came up with another different idea for this, will follow up